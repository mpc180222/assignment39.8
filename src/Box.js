import React, { useState } from 'react';
import './App.css';
import BoxList from './BoxList'


const Box = ({widthData, heightData, colorData, removeBox, key, id}) => {


    const boxStyle = {width: `${widthData}px`, height: `${heightData}px`, backgroundColor: colorData}

    return(
        <div>
        <div data-testid ={colorData} style = {boxStyle}>

        </div>
        <button data-testid={`remove-${colorData}`} id= {id} onClick = {removeBox}>X</button>
        </div>
    )


}

export default Box;

