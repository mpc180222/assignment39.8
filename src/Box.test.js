import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', () => {
  render(<Box widthData='100' heightData='100' colorData='orange'    />);
  
});

it("should match snapshot", () => {
  const { asFragment } = render(<Box />)
  expect(asFragment()).toMatchSnapshot();
})