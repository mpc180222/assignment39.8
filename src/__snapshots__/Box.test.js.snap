// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button
      data-testid="remove-undefined"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
